lib/matplotlib/artist.py

```diff
@@ -21,6 +21,31 @@
 _log = logging.getLogger(__name__)
 
 
+def _prevent_rasterization(draw):
+    # we assume that by default artists are not allowed to rasterize (unless
+    # its draw method is explicitly decorated). If it is being drawn after a
+    # rasterized artist and it has reached the rater_depth of 0. We stop
+    # rasterization so that it does not affect the behavior of normal artist
+    # (e.g., change in dpi). If the artist's draw method is decorated
+    # (draw._supports_rasterization is True), it won't  be decorated by
```

Comment:
But the check is not for `draw._supports_rasterization is True`; it's for `hasattr(draw, '_supports_rasterization')`

Reply:
I have changed the comment. 

` If the artist's draw method is decorated (has a "_supports_rasterization" attribute),`